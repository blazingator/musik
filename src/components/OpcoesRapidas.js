import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import { AntDesign } from '@expo/vector-icons'

export default function OpcoesRapidas({ iconName, iconColor, option }) {
  return (
    <TouchableOpacity style={styles.quickOption}>
      <AntDesign name={iconName} size={32} color={iconColor} />
      <Text style={styles.optionText}>{option}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  quickOption: {
    width: 60,
    height: 60,
    flex: 1,
    alignItems: 'center'
  },
  optionText: {
    fontSize: 12
  }
})
