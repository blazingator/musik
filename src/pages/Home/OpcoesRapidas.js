import React from 'react'
import { TouchableOpacity, Text,View, StyleSheet } from 'react-native'
import { Feather } from '@expo/vector-icons'

export default function OpcoesRapidas({ iconName, iconColor, option }) {
  return (
    <TouchableOpacity style={styles.quickOption}>
      <View style={styles.optionIcon}>
        <Feather name={iconName} size={32} color={iconColor} />
      </View>
      <Text style={styles.optionText}>{option}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  quickOption: {
    width: 70,
    height: 70,
    flex: 1,
    alignItems: 'center'
  },
  optionIcon: {
    width: 60,
    height: 60,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 3,
    borderRadius: 100,
    backgroundColor: '#fff'
  },
  optionText: {
    fontSize: 12
  }
})
