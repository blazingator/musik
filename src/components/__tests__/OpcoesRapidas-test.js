import React from 'react'
import OpcoesRapidas from '../OpcoesRapidas'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<OpcoesRapidas />).toJSON()
  expect(tree).toMatchSnapshot()
})
test('has at least 2 childs', () => {
  const tree = renderer.create(<OpcoesRapidas />).toJSON()
  expect(tree.children.length).toBeGreaterThanOrEqual(2)
})
