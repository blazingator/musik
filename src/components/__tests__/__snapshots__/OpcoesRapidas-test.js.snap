// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 60,
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
  />
</View>
`;
